Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc

```diff
@@ -52,11 +59,34 @@ bool HGCalParametersFromDD::build(const DDCompactView* cpv,
   DDSpecificsMatchesValueFilter filter{val};
   DDFilteredView fv(*cpv,filter);
   bool ok = fv.firstChild();
+  int  mode(0);
 
   if (ok) {
     DDsvalues_type sv(fv.mergedSpecifics());
-    php.mode_ = getGeometryMode("GeometryMode", sv);
+    php.mode_ = getGeometryMode("GeometryMode", sv, true);
```

Comment:
There are 2 types of enum's returned by getGeometryMode() - that is why i used the casting. In any case this is only when interpreting the xml file once in the beginning